### PR TITLE
Add docs for UC OSS for function calling

### PR DESCRIPTION
**Pull request by BenWilson2**
**PR Checklist**

- [X] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Adds docs and guides for UnityCatalog OSS AI function integration

### CODE REVIEW

File: docs/ai/client.md [hunk 1]
```diff
@@ -6,7 +6,194 @@ This guide provides detailed information about the Unity Catalog AI clients, inc
...
+from unitycatalog.ai.core.oss import UnitycatalogFunctionClient
+from unitycatalog.client import ApiClient, Configuration
+
+config = Configuration()
+config.host = "http://localhost:8080/api/2.1/unity-catalog"
+
+# The base ApiClient is async
+api_client = ApiClient(configuration=config)
+
+uc_client = UnitycatalogFunctionClient(uc=api_client)
```
Comment (BenWilson2) [inline docs/ai/client.md:66]:
```suggestion
uc_client = UnitycatalogFunctionClient(api_client=api_client)
```

File: docs/ai/integrations/anthropic.md [hunk 2]
```diff
@@ -38,7 +42,27 @@ pip install databricks-sdk "databricks-connect>=15.1.0"
...
+config = Configuration()
+# This is the default address when starting a UnityCatalog server locally. Update this to the uri
+# of your running UnityCatalog server.
+config.host = "http://localhost:8080/api/2.1/unity-catalog"
+
+# Create the UnityCatalog client
+api_client = ApiClient(configuration=config)
+
+# Use the UnityCatalog client to create an instance of the AI function client
+client = UnitycatalogFunctionClient(uc=api_client)
```
Comment (BenWilson2) [inline docs/ai/integrations/anthropic.md:62]:
```suggestion
client = UnitycatalogFunctionClient(api_client=api_client)
```

File: docs/ai/integrations/autogen.md [hunk 3]
```diff
@@ -42,7 +46,27 @@ To use Databricks-managed Unity Catalog with this package, follow the [Databrick
...
+config = Configuration()
+# This is the default address when starting a UnityCatalog server locally. Update this to the uri
+# of your running UnityCatalog server.
+config.host = "http://localhost:8080/api/2.1/unity-catalog"
+
+# Create the UnityCatalog client
+api_client = ApiClient(configuration=config)
+
+# Use the UnityCatalog client to create an instance of the AI function client
+client = UnitycatalogFunctionClient(uc=api_client)
```
Comment (BenWilson2) [inline docs/ai/integrations/autogen.md:66]:
```suggestion
client = UnitycatalogFunctionClient(api_client=api_client)
```

File: docs/ai/integrations/crewai.md [hunk 4]
```diff
@@ -42,7 +46,27 @@ To use Databricks-managed Unity Catalog with this package, follow the [Databrick
...
+config = Configuration()
+# This is the default address when starting a UnityCatalog server locally. Update this to the uri
+# of your running UnityCatalog server.
+config.host = "http://localhost:8080/api/2.1/unity-catalog"
+
+# Create the UnityCatalog client
+api_client = ApiClient(configuration=config)
+
+# Use the UnityCatalog client to create an instance of the AI function client
+client = UnitycatalogFunctionClient(uc=api_client)
```
Comment (BenWilson2) [inline docs/ai/integrations/crewai.md:66]:
```suggestion
client = UnitycatalogFunctionClient(api_client=api_client)
```

File: docs/ai/integrations/langchain.md [hunk 5]
```diff
@@ -36,7 +40,27 @@ pip install databricks-sdk "databricks-connect>=15.1.0"
...
+config = Configuration()
+# This is the default address when starting a UnityCatalog server locally. Update this to the uri
+# of your running UnityCatalog server.
+config.host = "http://localhost:8080/api/2.1/unity-catalog"
+
+# Create the UnityCatalog client
+api_client = ApiClient(configuration=config)
+
+# Use the UnityCatalog client to create an instance of the AI function client
+client = UnitycatalogFunctionClient(uc=api_client)
```
Comment (BenWilson2) [inline docs/ai/integrations/langchain.md:60]:
```suggestion
client = UnitycatalogFunctionClient(api_client=api_client)
```

File: docs/ai/integrations/llamaindex.md [hunk 6]
```diff
@@ -38,7 +42,27 @@ pip install databricks-sdk "databricks-connect==15.1.0"
...
+config = Configuration()
+# This is the default address when starting a UnityCatalog server locally. Update this to the uri
+# of your running UnityCatalog server.
+config.host = "http://localhost:8080/api/2.1/unity-catalog"
+
+# Create the UnityCatalog client
+api_client = ApiClient(configuration=config)
+
+# Use the UnityCatalog client to create an instance of the AI function client
+client = UnitycatalogFunctionClient(uc=api_client)
```
Comment (BenWilson2) [inline docs/ai/integrations/llamaindex.md:62]:
```suggestion
client = UnitycatalogFunctionClient(api_client=api_client)
```

File: docs/ai/integrations/openai.md [hunk 7]
```diff
@@ -36,7 +40,27 @@ pip install databricks-sdk "databricks-connect==15.1.0"
...
+config = Configuration()
+# This is the default address when starting a UnityCatalog server locally. Update this to the uri
+# of your running UnityCatalog server.
+config.host = "http://localhost:8080/api/2.1/unity-catalog"
+
+# Create the UnityCatalog client
+api_client = ApiClient(configuration=config)
+
+# Use the UnityCatalog client to create an instance of the AI function client
+client = UnitycatalogFunctionClient(uc=api_client)
```
Comment (BenWilson2) [inline docs/ai/integrations/openai.md:60]:
```suggestion
client = UnitycatalogFunctionClient(api_client=api_client)
```

File: docs/ai/quickstart.md [hunk 8]
```diff
@@ -53,7 +59,30 @@ seamlessly define a native LangChain tool that can interface with functions stor
...
+config = Configuration()
+# This is the default address when starting a UnityCatalog server locally. Update this to the uri
+# of your running UnityCatalog server.
+config.host = "http://localhost:8080/api/2.1/unity-catalog"
+
+# Create the UnityCatalog client
+api_client = ApiClient(configuration=config)
+
+# Use the UnityCatalog client to create an instance of the AI function client
+client = UnitycatalogFunctionClient(uc=api_client)
```
Comment (BenWilson2) [inline docs/ai/quickstart.md:79]:
```suggestion
client = UnitycatalogFunctionClient(api_client=api_client)
```

**Comment by serena-ruan** [comment]
@BenWilson2  Not related to this PR, but in case we forget, https://github.com/unitycatalog/unitycatalog/blob/main/ai/core/README.md needs to be updated as well :D

File: docs/ai/client.md [hunk 9]
```diff
@@ -6,7 +6,194 @@ This guide provides detailed information about the Unity Catalog AI clients, inc
...
+2. **Security Considerations**: Function execution occurs within the environment where a Unity Catalog function is called for execution. Be aware of the contents of any function prior to execution by checking the `routine_definition` entry by calling the `get_function` API if you did not author the function. Additionally, be aware that GenAI-generated Python code (for the use case of building a Unity Catalog function whose sole purpose is to execute arbitrary code) execution is inherently unsafe and should be performed in an isolated VM that is not connected to user data or is in an environment that is not ephemeral (a VM is ideal for this use case).
+3. **External Dependencies**: The environment that is executing the function that you're calling needs to have the appropriate packages pre-installed in order for functions that use external non-core-Python import statements. Ensure that dependent libraries are installed via `pip`. Additionally, import statements should be contained within the function body (using local imports) to ensure that the dependency is available for use by your function's internal logic.
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/ai/client.md:19]: Yep, this is going to be a proposal in Q1 for a server-based execution environment for python functions (design is still pending, but when it's available within Databricks, we'll be porting the solution over to UC OSS, using microVM process isolation and execution environment configuration options (i.e., I need the following pip packages installed in the environment). More to come on this later!

File: docs/ai/client.md [hunk 10]
```diff
@@ -6,7 +6,194 @@ This guide provides detailed information about the Unity Catalog AI clients, inc
...
+# The base ApiClient is async
+api_client = ApiClient(configuration=config)
+
+uc_client = UnitycatalogFunctionClient(uc=api_client)
+
+CATALOG = "my_catalog"
+SCHEMA = "my_schema"
+```
+
+> Tip: Ensure that you have created a catalog and a schema before attempting to create functions. The `ApiClient` via
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/ai/client.md]: Added the new APIs as reference from https://github.com/unitycatalog/unitycatalog/pull/765

File: docs/ai/client.md [hunk 11]
```diff
@@ -6,7 +6,194 @@ This guide provides detailed information about the Unity Catalog AI clients, inc
...
+2. **Security Considerations**: Function execution occurs within the environment where a Unity Catalog function is called for execution. Be aware of the contents of any function prior to execution by checking the `routine_definition` entry by calling the `get_function` API if you did not author the function. Additionally, be aware that GenAI-generated Python code (for the use case of building a Unity Catalog function whose sole purpose is to execute arbitrary code) execution is inherently unsafe and should be performed in an isolated VM that is not connected to user data or is in an environment that is not ephemeral (a VM is ideal for this use case).
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/ai/client.md]: good idea!

**Pull request merged**